src/components/Navigation.jsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import styles from "./Navigation.module.css";
+import {NavLink, useNavigate} from "react-router-dom";
+
+const Navigation = ({auth, setAuth}) => {
+
+    const navigate = useNavigate()
+
+    return (
+        <>
+            <nav className={styles["navigationContainer"]}>
+                <ul className={styles["navigationContent"]}>
+                    <li className={styles["navigationItem"]}>
+                        <NavLink
+                            className={({isActive}) => isActive ? 'active-menu-link' : 'default-menu-link'}
+                            to={"/"}>
+                            Home
+                        </NavLink>
+                    </li>
+                    <li className={styles["navigationItem"]}>
+                        <NavLink
+                            className={({isActive}) => isActive ? 'active-menu-link' : 'default-menu-link'}
+                            to={"/login"}>
+                            Login
+                        </NavLink>
+                    </li>
+                    {auth && <li className={styles["navigationItem"]}>
+                        <NavLink
+                            className={({isActive}) => isActive ? 'active-menu-link' : 'default-menu-link'}
+                            to={"/blogposts"}>
+                            Blog overzicht
+                        </NavLink>
+                    </li>}
+                    {auth && <li className={styles["navigationItem"]}>
+                        <NavLink
+                            className={({isActive}) => isActive ? 'active-menu-link' : 'default-menu-link'}
+                            to={"/blogposts/:blogId"}>
+                            Blog post
+                        </NavLink>
```

Comment:
Deze `NavLink` geeft een error, omdat je naar een blogpost linkt maar geen ID doorgeeft via de URL. Hierdoor is je ID undifined en kan hij de data niet laden.  Deze link zou ik hier ook weghalen of wel een id meegeven, dat je naar 1 bepaalde blogpost doorlinkt.